Source/engine.h

```diff
@@ -137,7 +137,14 @@ struct CelOutputBuffer {
 	 */
 	CelOutputBuffer subregion(Sint16 x, Sint16 y, Uint16 w, Uint16 h) const
 	{
-		return CelOutputBuffer(surface, SDL_Rect { region.x + x, region.y + y, w, h });
+		// In SDL1 SDL_Rect x and y are Sint16. Cast explicitly to avoid a compiler warning.
+		using CoordType = decltype(SDL_Rect {}.x);

```

Comment:
Oh wow... is this the equivalent of `typeof` in C#?

Reply:
I don't know C# but `decltype` returns the type of any given expression without evaluating it